compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -3680,6 +3680,19 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                         Applicability::MachineApplicable,
                     );
                 }
+                // note if the user wants to use associated items of a trait directly
+                if in_path && res.len() == 0 {
```

Comment:
Why are we checking `res` here?

Reply:
Because I think there is no need to make this note for trait objects with multi traits such as `<TraitA + TraitB>::foo()`